common/test/acceptance/tests/lms/test_teams.py

```diff
@@ -1105,6 +1108,107 @@ def test_discussion_privileged_user_can_edit_team(self, role):
+        Given I am staff user for a course with a team
+        When I visit the Team profile page
+        Then I should see the Edit Team button
+        And When I click edit team button
+        Then I should see the Edit Membership button
+        And When I click the edit membership button
+        Then I should see the edit membership page
+        And When I click the remove button and confirm the dialog
+        Then my membership should be removed, and I should remain on the page
+        """
+        user_info = AutoAuthPage(
+            self.browser,
+            course_id=self.course_id,
+            staff=False,
+            roles=role
+        ).visit().user_info
+
+        self.create_membership(user_info['username'], self.team['id'])
+
+        self.team_page.visit()
+        self.team_page.click_edit_team_button()
+
+        #TODO: update name of create_edit_team_page here too
+        self.create_edit_team_page.wait_for_page()
+        self.assertTrue(
+            self.create_edit_team_page.membership_button_present
+        )
+
+        self.create_edit_team_page.click_membership_button()
+        self.edit_membership_page.wait_for_page()
+        self.assertEqual(self.edit_membership_page.team_members, 1)
+        self.edit_membership_page.click_first_remove()
+        self.edit_membership_page.confirm_delete_membership_dialog()
+        self.assertEqual(self.edit_membership_page.team_members, 0)
+        self.assertTrue(self.edit_membership_page.is_browser_on_page)
+
+    @ddt.data('Moderator', 'Community TA', 'Administrator')
+    def test_cancel_remove_membership(self, role):
+        """
+        Scenario: The user should be able to remove a membership
+        Given I am staff user for a course with a team
+        When I visit the Team profile page
+        Then I should see the Edit Team button
+        And When I click edit team button
+        Then I should see the Edit Membership button
+        And When I click the edit membership button
+        Then I should see the edit membership page
+        And When I click the remove button and cancel the dialog
+        Then my membership should not be removed, and I should remain on the page
+        """
+        user_info = AutoAuthPage(
+            self.browser,
+            course_id=self.course_id,
+            staff=False,
+            roles=role
+        ).visit().user_info
+
+        self.create_membership(user_info['username'], self.team['id'])
+
+        self.team_page.visit()
+        self.team_page.click_edit_team_button()
+
+        #TODO: update name of create_edit_team_page here too
+        self.create_edit_team_page.wait_for_page()
+        self.assertTrue(
+            self.create_edit_team_page.membership_button_present
+        )
+
+        self.create_edit_team_page.click_membership_button()
+        self.edit_membership_page.wait_for_page()
+        self.edit_membership_page.click_first_remove()
```

Comment:
I would create a helper method for all the setup that is in common with the previous test case.